### PR TITLE
fix(MessageButtonsBar): pass props only

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Related to #12380 and to avoid passing class as a prop in vue 3 

Tested with Vue3:
image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/Message.vue [hunk 1]
```diff
@@ -440,6 +440,23 @@ export default {
...
+				timestamp: this.timestamp,
+				id: this.id,
+				isReplyable: this.isReplyable,
+				token: this.token,
+				messageType: this.messageType,
+				previousMessageId: this.previousMessageId,
+				lastEditActorDisplayName: this.lastEditActorDisplayName,
+				lastEditTimestamp: this.lastEditTimestamp,
+			}
+		},
```
Comment (ShGKme) [inline src/components/MessagesList/MessagesGroup/Message/Message.vue:458]: If we already consider it an atomic object, then it makes sense to pass it as a single prop.

Comment (DorraJaouad) [reply src/components/MessagesList/MessagesGroup/Message/Message.vue:458]: Not really, there is `isReplyable` and it doesn't cover the whole message object that we commonly use.

Comment (Antreesy) [reply src/components/MessagesList/MessagesGroup/Message/Message.vue:458]: > there is `isReplyable`

Which is a part of message object. The only thing missing from it is `previousMessageId`, and its usage looks minor, so I'd pass it separately and think of a way to not use it in the component at all (by modifying store action?)

Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/Message.vue:458]: If we consider some specific props to not be a part of the `message` object we can pass them separately, keeping all the other properties a part of the `message` object.

So, if we want to refactor it, let's in both `Message` and `MessageButtonsBar` add a prop for `message` object with all the properties from the original `message` object (from store = API) and keep only additional properties as separated props.

Comment (Antreesy) [reply src/components/MessagesList/MessagesGroup/Message/Message.vue]: Refactored

**Comment by DorraJaouad** [comment]
Changes lGTM 👍🏻

**Review by Antreesy — APPROVED** [review]
Approving on behalf of Dorra :eagle: . Tested:
- readInfo properties on children
- callback actions (Deck app)
- polls, replies, markdown, failures, captions, system messages, edited / deleted messages. rich params, buttonbar actions

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/MessageBody.vue [hunk 2]
```diff
@@ -156,101 +156,36 @@ export default {
...
 	},
 
 	setup(props) {
-		const { token, id } = toRefs(props)
+		const { message } = toRefs(props)
 		const {
 			isEditable,
 			isFileShare,
-		} = useMessageInfo(token, id)
+		} = useMessageInfo(message.value.token, message.value.id)
```
Comment (ShGKme) [inline src/components/MessagesList/MessagesGroup/Message/MessagePart/MessageBody.vue:188]: This is reactivity loosing

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 3]
```diff
@@ -477,7 +397,7 @@ export default {
 			isFileShareWithoutCaption,
 			isConversationReadOnly,
 			isConversationModifiable,
-		} = useMessageInfo(token, id)
+		} = useMessageInfo(message.value.token, message.value.id)
```
[the remark this answers is not in the document]
Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:400]: Still reactivity loosing

[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:400]: It wasn't reactive initially.. 
But now we could passed a whole message object and make token and I'd truly reactive

**Pull request merged**